src/redux/reducer.js

```diff
@@ -0,0 +1,107 @@
+import {createSlice} from "@reduxjs/toolkit";
+import {getDataInRange, getEndpoints, getServices, getServicesName,} from "../api";
+import {MONTH, WEEK} from "../tools/constant";
+import {formattedDate} from "../tools/FomatedDate";
+
+const chartSlice = createSlice({
```

Comment:
let's get rid from redux, looks too complex for simple grid (use https://reactjs.org/docs/lifting-state-up.html if required), right?